### PR TITLE
feat: fueling and boarding

### PR DESCRIPTION
**Pull request by chloe1192**
* Adds a convenient way to load fuel and payload via the CDU
* Adds a feature that saves the amount of fuel, so that the amount of fuel upon starting the next flight is whatever it was when the last flight ended.

image
image

**Testing instructions**
* Test the fueling and boarding features extensively, manually inputting values as well as importing from SimBrief, using both kg and lbs units. Make sure the fuel and weight is distributed in a reasonable way. Test out the different rate options. **Note: make sure to use the [new SimBrief airframe](https://www.simbrief.com/system/dispatch.php?sharefleet=141372_1660066082616).**

### CODE REVIEW

**Comment by DesertRick13** [comment]
If this happens, I hope there is an option for a 747-8F, BCF, and Passenger.

**Comment by iamdumdum1234** [comment]
Very cool!

File: salty-747/html_ui/Pages/Salty/SaltyStates.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        this.main4stored = SaltyDataStore.get("747_MAIN4_LAST_QUANTITY", '') ? parseFloat(SaltyDataStore.get("747_MAIN4_LAST_QUANTITY", '')) : "";
+        this.res1stored = SaltyDataStore.get("747_RES1_LAST_QUANTITY", '') ? parseFloat(SaltyDataStore.get("747_RES1_LAST_QUANTITY", '')) : "";
+        this.res2stored = SaltyDataStore.get("747_RES2_LAST_QUANTITY", '') ? parseFloat(SaltyDataStore.get("747_RES2_LAST_QUANTITY", '')) : "";
+        this.centerstored = SaltyDataStore.get("747_CENTER_LAST_QUANTITY", '') ? parseFloat(SaltyDataStore.get("747_CENTER_LAST_QUANTITY", '')) : "";
+        this.stabstored = SaltyDataStore.get("747_STAB_LAST_QUANTITY", '') ? parseFloat(SaltyDataStore.get("747_STAB_LAST_QUANTITY", '')) : "";
+    }// ends constructor
+
+    Init() {
+
+    }
```
Comment (ninjomcs) [inline salty-747/html_ui/Pages/Salty/SaltyStates.js:19]: cleaner to get rid of this if it's empty

File: salty-747/html_ui/Pages/Salty/SaltyStates.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                SaltyDataStore.set("747_CENTER_LAST_QUANTITY", centerCurrentSimVar.toString());
+                SaltyDataStore.set("747_STAB_LAST_QUANTITY", stabCurrentSimVar.toString());
+                console.log(SaltyDataStore.get("747_MAIN1_LAST_QUANTITY") + " main1");
+            }
+        }// ends if flighthasstarted
+    }// ends onflightstart
+
+    update(_deltaTime) {
+
+    }// ends update
```
Comment (ninjomcs) [inline salty-747/html_ui/Pages/Salty/SaltyStates.js:66]: same as above

File: salty-747/html_ui/Pages/VCockpit/Instruments/Airliners/B747_8/FMC/B747_8_FMC_Field.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
+class FMC_Field {
```
Comment (ninjomcs) [inline salty-747/html_ui/Pages/VCockpit/Instruments/Airliners/B747_8/FMC/B747_8_FMC_Field.js:1]: what exactly is the purpose of this class / what does it do?

**Review by ninjomcs — CHANGES_REQUESTED** [review]

**Comment by giridat** [comment]
Name: Giridhar Raut#5534
Date of testing: 14/02/2022
Version of the sim: Latest
PR Tested: #339 

Testing technique: Normal spawn C&D
Steps you've made to test the Pull Request:
1. Turn on acft go into boarding options 
2. Press FMC
3. FMC Comm > Boarding
4. Tested each page with different values and all at all the rates (Real, Fast, Instant)

Overall Rating: To be improved/ NOT PASS (check out the videos below)
Conclusions of the testing: 

1. The refueling option once started doesn't stop automatically after the refueling is complete whereas the other boarding options stop automatically.
2. I expect a 'Reset all Values to default' button which would reset the aircraft cargo and pax state to what it was upon spawning.
3. It would be feasible to know what unit KG or LBS is being used on the cargo page. 
4. I also want to know why the total PAX capacity is going to 364 whereas IRL it is 467?
5. The passenger page can be broken easily as the boarding rate for the pax page is on the previous Cargo page due to which if you change the rate on the cargo page which I believe changes the rate for the pax page as well sometimes doesn't happen video has been included for the same pax_page_behaviour.mp4.


https://user-images.githubusercontent.com/61822722/153805119-fe0faede-2673-4bc5-8ba0-7b535f3f5024.mp4


https://user-images.githubusercontent.com/61822722/153805131-007bbecf-6dd8-4cf7-b6b7-236b0062f5fc.mp4


https://user-images.githubusercontent.com/61822722/153805135-98c0e4ad-495b-40aa-aeaa-4a35d477446c.mp4



https://user-images.githubusercontent.com/61822722/153805326-30a59c80-9672-45f8-b3bd-28da7e5fb239.mp4

**Review by ninjomcs — APPROVED** [review]

**Comment by brewers10** [comment]
Name: brewers_10#5706
Date of testing:  11/08/2022
Version of the sim: Latest SU9 version
PR Tested: #339 

Testing technique: Loaded into the Salty 747 mod. Simbrief flight from YLRE --> YSSY created (units in KG to start) and loaded into the FMC. Loaded the route, fuel and payload all successfully and matching the values in the OFP. Switched units in FPL to LBS and also switched the units in the Salty 747 to LBS. Reloaded FPLN into the FMC and re-requested fuel and payload data aswell... all loaded in without any issues: very convenient. 
Steps you've made to test the Pull Request:
1. Loaded into YLRE in the Salty 747 (Salty Livery)
2. Loaded OFP into FMC
3. Requested Payload and Fuel figures from Simbrief
4. Loaded all payload and fuel onto the aircraft.
5. Repeated with units in LBS and also manually inputted figures to be loaded. No issues

Overall Rating: Good
Conclusions of the testing: It's hard to explain, but when requesting payload figures from Simbrief. The "Request send" button remains static, normally, when you request something the "Send" disappears whilst the aircraft retrieves data from Simbrief. If you need that explaining better contact me on Discord :)

**Comment by oz2010** [comment]
Name: oz2010#1558
Date of testing: 11/08/2022
Version of the sim: Latest SU9 version
PR Tested: https://github.com/saltysimulations/salty-747/pull/339

Testing technique: Load flight EDDF->KLAX, 316 pax 122.9k kg fuel ~6.3k kg payload. Try OFP request as well as manually setting all values. Messed around with the load speeds (fast/real/instant) to see how they work, also did some limits testing (details below)

Steps you've made to test the Pull Request:
1. Load flight as specified above
2. Do OFP request for the flight, start loading fuel and pax/cargo. Payload manager is able to handle both simultaneously as should be expected. Hit stop/start intermittently on both to check it stops and starts. Loading speed can be changed while stopped, and will resume at the new desired speed when started again.
3. Changed loading speed while loading, the effect is instant. Notable that changing to instant speed while loading will instantly finish.
4. Tried setting pax above the limit of 364, said "not allowed" as expected
5. Manually set desired fuel value to 0 and watched fuel tanks drain. Inner tanks drain first, followed by outer (as expected I believe). Then did OFP request again and filled back up to flight fuel
6. Tried higher values of fuel until it stopped letting me. The max appears to be 191.0k kg (technically 191.06 is the highest, which displays rounded as 191.1k kg). Interestingly, setting 191.1kg (or anything up to 191.19) shows "not allowed" on the scratchpad, but any higher values do not (instead just have no effect)


Overall Rating: Good
Conclusions: Great payload manager. Any issues are really small and nitpicky
1. For example, everything displays as "kg" but a way to show it's thousands of kg would be nice, but not strictly necessary (I recall the heavy division 787 does this too).
2. Cycling through the loading speeds is real->fast->instant->back to real, etc. ie no way to go backwards, so if you're trying to switch fast to real while actively loading you can't (it will instead switch to instant which instantly finishes). Not a big deal since the fix is just to stop, change speed to real by clicking twice, then start again - but it did catch me off guard while messing with stuff

Cheers to saying goodbye to the MSFS payload manager!

**Review by ninjomcs — APPROVED** [review]

**Review by ninjomcs — APPROVED** [review]

**Comment by Quantados** [comment]
Name: Quantados#7632
Date of testing: Current date 13/08/2022
Version of the sim: SU10 Beta 3
PR Tested: #339 

Testing technique: Load into aircraft, loaded KSFO-KBOS route. Loaded fuel, payload, and route. Everything loaded correctly without any problems. 

Steps you've made to test the Pull Request:
1. Load into KSFO.
2. Requested OFP into fuel, payload, and route menus.
3. Changed Salty 747 units from LBS to KG a couple times.
4. Loaded fuel and payload.
5. Tried loading with real, fast, and instant rates.
6. Manually set fuel 10,000 lbs higher then tried draining it back to original.

Overall Rating: Good

Conclusions of the testing: 
Everything looked good. One small thing I saw was that SEL Fuel/SEL Cargo doesn't automatically convert when changing units in Salty 747.

**Pull request merged**